scripts/build_macos.sh

```
@@ -6,16 +6,37 @@ IDF_VERSION="${IDF_VERSION:-v5.5.2}"
+  done
+  echo "Python >= 3.10 not found. Run scripts/setup_idf_macos.sh first." >&2
+  return 1
+}
 
 if [[ ! -f "$IDF_DIR/export.sh" ]]; then
   echo "ESP-IDF not found at: $IDF_DIR" >&2
   echo "Run scripts/setup_idf_macos.sh first, or set IDF_DIR/IDF_PATH." >&2
   exit 1
 fi
 
+export ESP_PYTHON="$(pick_python)"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)